### PR TITLE
adds user tag functionality to Metadata tab

### PR DESCRIPTION
**Pull request by astewes**
-  Adds user/preferred tags
-  Adds ability to delete these tags
-  Adds functionality to add user/preferred tag to an entity
-  Adds typeahead functionality to populate menu of tags in use

### CODE REVIEW

File: cdap-ui/app/features/tracker/services/my-tracker-api.js [hunk 1]
```diff
@@ -67,7 +67,10 @@ function myTrackerApi(myCdapUrl, $resource, myAuth, myHelpers, UI_CONFIG) {
     demotePreferredTags: myHelpers.getConfig('POST', 'REQUEST', tagsPath + '/demote', false, { suppressErrors: true }),
     promoteUserTags: myHelpers.getConfig('POST', 'REQUEST', tagsPath + '/promote', false, { suppressErrors: true }),
     validatePreferredTags: myHelpers.getConfig('POST', 'REQUEST', tagsPath + '/validate', false, { suppressErrors: true }),
+    getEntityTags: myHelpers.getConfig('GET', 'REQUEST', tagsPath + '/:entityType/:entityId', false, { suppressErrors: true }),
+    addEntityTag: myHelpers.getConfig('POST', 'REQUEST', tagsPath + '/promote/:entityType/:entityId', false, { suppressErrors: true }),
```
Comment (elfenheart) [inline cdap-ui/app/features/tracker/services/my-tracker-api.js]: adding entity tag does not mean promoting it as a preferred tag.. so calling promote here is wrong

File: cdap-ui/app/features/tracker/templates/metadata.html [hunk 2]
```diff
@@ -25,48 +25,81 @@
   <!-- TAGS SIDEBAR -->
   <div class="entity-detail-tags-column">
     <div class="entity-container">
-      <div class="user-tags" ng-if="MetadataController.tags.user.length > 0">
+      <div class="user-tags" ng-if="MetadataController.tags.userTags.length > 0 || MetadataController.tags.preferredTags.length > 0">
```
Comment (elfenheart) [inline cdap-ui/app/features/tracker/templates/metadata.html]: this condition is wrong. If there is no userTags or preferredTags, it should still show the plus button

File: cdap-ui/app/features/tracker/controllers/metadata-ctrl.js [hunk 3]
```diff
@@ -208,6 +217,133 @@ class TrackerMetadataController {
...
+      });
+  }
+
+  goToTag(event, tag) {
+    event.stopPropagation();
+    this.$state.go('search.objectswithtags', {tag: tag});
+  }
+
+  escapeInput() {
+    this.$scope.model = '';
```
Comment (elfenheart) [inline cdap-ui/app/features/tracker/controllers/metadata-ctrl.js]: any particular reason why the model is in $scope property?

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by elfenheart** [comment]
👍  LGTM

**Comment by prinam** [comment]
Is there a link to the RAT check build?

**Comment by astewes** [comment]
Bamboo build: http://builds.cask.co/browse/CDAP-DRC4085-3

**Pull request merged**